task/sast-shell-check/0.1/sast-shell-check.yaml

```diff
@@ -0,0 +1,211 @@
+        - name: KFP_GIT_URL
+          value: $(params.KFP_GIT_URL)
+        - name: PROJECT_NAME
+          value: $(params.PROJECT_NAME)
+        - name: RECORD_EXCLUDED
+          value: $(params.RECORD_EXCLUDED)
+        - name: IMP_FINDINGS_ONLY
+          value: $(params.IMP_FINDINGS_ONLY)
+      script: |
+        #!/usr/bin/env bash
+        set -x
+        # shellcheck source=/dev/null
+        source /utils.sh
+        trap 'handle_error $(results.TEST_OUTPUT.path)' EXIT
+
+        ca_bundle=/mnt/trusted-ca/ca-bundle.crt
+        if [ -f "$ca_bundle" ]; then
+          echo "INFO: Using mounted CA bundle: $ca_bundle"
+          cp -vf $ca_bundle /etc/pki/ca-trust/source/anchors
+          update-ca-trust
+        fi
+
+        PACKAGE_VERSION=$(rpm -q --queryformat '%{NAME}-%{VERSION}-%{RELEASE}\n' ShellCheck)
+
+        OUTPUT_FILE="shellcheck-results.json"
+        SOURCE_CODE_DIR=$(workspaces.workspace.path)/source
+
+        # generate all shellcheck result JSON files to $SC_RESULTS_DIR, which defaults to ./shellcheck-results/
+        /usr/share/csmock/scripts/run-shellcheck.sh "$SOURCE_CODE_DIR"
+
+        CSGREP_OPTS=(
+            --mode=json
+            --strip-path-prefix="$SOURCE_CODE_DIR"/
+            --remove-duplicates
+            --embed-context=3
+            --set-scan-prop="ShellCheck:${PACKAGE_VERSION}"
+        )
+        if [[ "$IMP_FINDINGS_ONLY" == "true" ]]; then
+            # predefined list of shellcheck important findings
+            CSGREP_EVENT_FILTER='\[SC(1020|1035|1054|1066|1068|1073|1080|1083|1099|1113|1115|1127|1128|1143|2043|2050|'
+            CSGREP_EVENT_FILTER+='2055|2057|2066|2069|2071|2077|2078|2091|2092|2157|2171|2193|2194|2195|2215|2216|'
+            CSGREP_EVENT_FILTER+='2218|2224|2225|2242|2256|2258|2261)\]$'
+            CSGREP_OPTS+=(
+                --event="$CSGREP_EVENT_FILTER"
+            )
+        else
+            CSGREP_OPTS+=(
+                --event="error|warning"
+            )
+        fi
+
+        if ! csgrep "${CSGREP_OPTS[@]}" ./shellcheck-results/*.json > "$OUTPUT_FILE"; then
+            echo "Error occurred while running 'run-shellcheck.sh'"
+            note="Task $(context.task.name) failed: For details, check Tekton task log."
+            ERROR_OUTPUT=$(make_result_json -r ERROR -t "$note")
+            echo "${ERROR_OUTPUT}" | tee "$(results.TEST_OUTPUT.path)"
+            exit 1
+        fi
+
+        # Filter known false positives if KFP_GIT_URL is set
+        if [ -n "${KFP_GIT_URL}" ]; then
+            echo "Filtering known false positives using ${KFP_GIT_URL}"
+
+            # build initial csfilter-kfp command
+            csfilter_kfp_cmd=(
+                csfilter-kfp
+                --verbose
+                --kfp-git-url="${KFP_GIT_URL}"
+            )
+
+            # Append --project-nvr option if PROJECT_NAME is set
+            if [ -n "${PROJECT_NAME}" ]; then
+                csfilter_kfp_cmd+=(--project-nvr="${PROJECT_NAME}")
```

Comment:
@rhyw What we did in other SAST tasks is that we have added a new parameter called `COMPONENT`, which is taken from the metadata labels: https://github.com/konflux-ci/build-definitions/blob/main/task/sast-snyk-check/0.3/sast-snyk-check.yaml#L98-L101 . That parameter is always used unless the user adds a `PROJECT_NAME` : https://github.com/konflux-ci/build-definitions/blob/main/task/sast-snyk-check/0.3/sast-snyk-check.yaml#L110-L113 . Would you mind updating the PR?

Reply:
Updated, thanks!